packages/react-compose/src/next/simplifyShorthand.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { GenericDictionary } from './types';
+
+/**
+ * Ensures that the given slots are represented using object syntax. This ensures that
+ * the object can be merged along with other objects.
+ * @param props - The incoming props
+ * @param shorthandPropNames - An array of prop names to apply simplification to
+ */
+export const simplifyShorthand = (props: GenericDictionary, shorthandPropNames?: string[]) => {
+  let newProps = props;
+
+  if (shorthandPropNames && shorthandPropNames.length) {
+    newProps = {
+      ...props,
+    };
+    for (const propName of shorthandPropNames) {
+      const propValue = props[propName];
+
+      if (propValue !== undefined && (typeof propValue !== 'object' || React.isValidElement(propValue))) {
+        newProps[propName] = { children: propValue };
```

Comment:
v0 shorthands can map to a different `prop` based on config:
https://github.com/microsoft/fluentui/blob/10a4c280bd1feb94a149e35555d0539d55ff24a8/packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx#L579-L581
and also supports an array collections:
https://github.com/microsoft/fluentui/blob/10a4c280bd1feb94a149e35555d0539d55ff24a8/packages/fluentui/react-northstar/src/components/Menu/Menu.tsx#L389-L391

Do you plan to support these as well?

Reply:
Yep! I haven't yet added this;

Reply:
Will leave this in a separate PR. I've made a note for pending change.